plugin/src/main/java/org/opennms/plugins/cloud/tsaas/GrpcExceptionHandler.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.plugins.cloud.tsaas;
+
+import static io.grpc.Status.Code.DEADLINE_EXCEEDED;
+import static io.grpc.Status.Code.OK;
+import static io.grpc.Status.Code.UNAVAILABLE;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.opennms.integration.api.v1.timeseries.StorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.grpc.StatusRuntimeException;
+
+/**
+ * Tries to distinguish between
+ * - recoverable Exceptions: will be propgated to OpenNMS and
+ * - non recoverable Exceptions: will be logged dropped
+ * see also: https://www.grpc.io/docs/guides/error/
+ * */
+public class GrpcExceptionHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GrpcExceptionHandler.class);
+  private static final Set<Code> RECOVERABLE_EXCEPTIONS = new HashSet<>(Arrays.asList(DEADLINE_EXCEEDED, UNAVAILABLE));
```

Comment:
@jbeuree do you think these are the right recoverable exceptions? Sere also: https://www.grpc.io/docs/guides/error/

Reply:
Might make sense to also include the UNAUTHENTICATED error as well - don't drop any outgoing data while we wait for the credentials to be corrected. This should be the error that occurs if we revoke the cert and force them to get a new one.